src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/TargetedDependenciesSnapshot.cs

```diff
@@ -58,7 +58,7 @@ internal TargetedDependenciesSnapshot(IDictionary<string, IDependency> dependenc
 
         private readonly Dictionary<string, IDependency> _topLevelDependenciesByPathMap = new Dictionary<string, IDependency>(StringComparer.OrdinalIgnoreCase);
         private readonly ConcurrentDictionary<string, IList<IDependency>> _dependenciesChildrenMap = new ConcurrentDictionary<string, IList<IDependency>>(StringComparer.OrdinalIgnoreCase);
-        private readonly Dictionary<string, bool> _unresolvedDescendantsMap = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+        private readonly ConcurrentDictionary<string, bool> _unresolvedDescendantsMap = new ConcurrentDictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Also, adding this comment again in case you don't see the Outdated one:

Unless this was a known contentious path, this may not be worth it. In any case, you might want to limit the concurrencyLevel of the new dictionary, since ConcurrentDictionary tends to be a very large allocation by default.

Reply:
You're right. It prevents the corruption. The race isn't such an issue as it would only result in duplicated work. But actually it's not hard to address that too. I've pushed a change that uses `ConcurrentDictionary<,>.GetOrAdd`.